### PR TITLE
 Upgrade guide for 'kubernetes_manifest' resources.

### PR DESCRIPTION
**Pull request by alexsomesan**
### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
...
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: website/docs/guides/alpha-manifest-migration-guide.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+# Use Case A
+
+## Using only `kubernetes_manifest` resources
+
+If you've already been using the experimental `kubernetes-alpha` provider, you are likely falling in this use-case. If you have other `kubernetes_*` resources in your configuration, please also see [Use-case B](#use-case-b)
+
+In case your configuration only uses `kubrenetes_manifest`, but no other resources of the Kubernetes provider (e.g. `kubernetes_*` resources) follow these steps to migrate.
+
+**Make sure you don't have any existing state generated with the `kubernetes-alpha` provider!**
```
Comment (redeux) [inline website/docs/guides/alpha-manifest-migration-guide.markdown:24]: I'm worried this requires a depth of knowledge many users won't have. Can you also describe how to check that they isn't any existing state generated with the `kubernetes-alpha` provider?

Comment (alexsomesan) [reply website/docs/guides/alpha-manifest-migration-guide.markdown:24]: Good point! Let me add that.

**Review by redeux — APPROVED** [review]
Looks great!

**Review by jrhouston — APPROVED** [review]
Good to merge once the website linter is passing

File: website/docs/guides/alpha-manifest-migration-guide.markdown [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+## Mixing 'kubrenetes_manifest' with other 'kubernetes_*' resources
+
+In case you plan on adding `kubrenetes_manifest` resources to your existing configuration which contains other resources of the Kubernetes provider there are some important aspects to be aware of.
+
+If your present configuration for the Kubernetes provider also creates the Kubernetes cluster using Terraform resources in the same `apply` operation (against best-practice recommendations), this will no longer work when adding `kubrenetes_manifest` resources. The reason behind this is that `kubrenetes_manifest` require access to the API during planning, at which point the cluster resource would not have yet been created.
+
+As a solution, choose one of the following options:
+
+* separate the cluster creation in the a different `apply` operation.
+* add a new `apply` operation only for the `kubernetes_manifest` resources.
```
Comment (eljefedelrodeodeljefe) [inline website/docs/guides/alpha-manifest-migration-guide.markdown:118]: @alexsomesan This specific behaviour poses some significant challenges for some processes e.g. on GCP / GKE creating cdn proxies via three resources but 6 referenced ones e.g. other services and pods.

Has there been any discussion to enhance the behaviour here e.g. not requiring client during planning?

**Pull request merged**